### PR TITLE
projects/cincinnati: reconfigure namespace to build base image

### PR DESCRIPTION
**Pull request by vrutkovs**
This PR updates `cincinnati` namespace to make e2e tests reuse builder image.

* Updates OWNERS for this namespace
* Adds rbac rules so that we could control builds there

Ref: https://issues.redhat.com/browse/OTA-147

### CODE REVIEW

File: projects/cincinnati/OWNERS [hunk 1]
```diff
@@ -0,0 +1,39 @@
+approvers:
+- aditya-konarde
+- crawford
+- jakedt
+- jfchevrette
+- jmelis
+- jottofar
+- jzelinskie
+- lalatendumohanty
+- lucab
```
Comment (steveeJ) [inline projects/cincinnati/OWNERS:10]: Luca removed himself from the Cincinnati OWNERS files recently, we can skip him here as well.

File: projects/cincinnati/OWNERS [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+reviewers:
+- aditya-konarde
+- crawford
+- jakedt
+- jfchevrette
+- jmelis
+- jottofar
+- jzelinskie
+- lalatendumohanty
+- lucab
```
Comment (steveeJ) [inline projects/cincinnati/OWNERS:30]: same as above

File: projects/cincinnati/OWNERS (hunk 1 above)
Comment (vrutkovs) [reply projects/cincinnati/OWNERS]: Fixed

**Comment by vrutkovs** [comment]
>Does this mean that we'll be manually updating the build root image?

(that's for https://github.com/openshift/release/pull/8186 really)
Yes, at this point we'd trigger buildconfig manually whenever base image changes are committed in master

**Comment by steveeJ** [comment]
>  Yes, at this point we'd trigger buildconfig manually whenever base image changes are committed in master

What's the mechanism to trigger it? Are base image changes still verified by CI?

**Comment by vrutkovs** [comment]
>What's the mechanism to trigger it?

`oc` command or web console triggering

> Are base image changes still verified by CI?

There are no tests for it, it cannot be verified by itself. There are no tests the cincinnati would be rebuilt using it and passes the e2e.

However in cincinnati definition we could specify a particular image we'd like to use as a base - and bump it via release repo PRs

**Comment by steveeJ** [comment]
> However in cincinnati definition we could specify a particular image we'd like to use as a base - and bump it via release repo PRs

I'd like to go through the workflow before we introduce a manual step here. IIUC we want to bump this with at least every new Rust release, correct? Ultimately this image would be shared between CI, stage and production builds so as to minimize the time from commit to production rollout.

**Comment by vrutkovs** [comment]
>IIUC we want to bump this with at least every new Rust release, correct? 

Right

>Ultimately this image would be shared between CI, stage and production builds so as to minimize the time from commit to production rollout.

Correct at the moment, however:
a) stage and prod builds would need to be updated, as they currently do a full rebuild
b) production builds would (eventually) use RHEL-based images instead of CentOS ones

**Comment by steveeJ** [comment]
> Correct at the moment, however:
> a) stage and prod builds would need to be updated, as they currently do a full rebuild
> b) production builds would (eventually) use RHEL-based images instead of CentOS ones

Thanks. Given this outlook I would prefer to skip the root caching for now. Instead, let's direct out attention towards unifying the build environment around the RHEL-based production builds.

**Comment by steveeJ** [comment]
>  a) its not public yet, https://github.com/sclorg/rust-container is 1.20

Both of that can be changed.





>  b) we'd still need additional tools like kcov there

Are there other projects which have solved the problem of having an automatically managed, cached, build image on top of an official builder? It's surprising that there's no off-the-shelve solution for this use-case.

**Comment by vrutkovs** [comment]
>Are there other projects which have solved the problem of having an automatically managed, cached, build image on top of an official builder? It's surprising that there's no off-the-shelve solution for this use-case.

Golang-based project use versioned images (`registry.svc.ci.openshift.org/openshift/release:golang-1.13`) and so could we - e.g. `cincinnati:rust-1.40`)

Again, this is the wrong PR to discuss this -> https://github.com/openshift/release/pull/8186

File: projects/cincinnati/OWNERS [hunk 3]
```diff
@@ -0,0 +1,37 @@
+approvers:
+- aditya-konarde
```
Comment (wking) [inline projects/cincinnati/OWNERS:2]: Pretty sure app-SRE doesn't care about what's happening over in openshift/release.  @aditya-konarde , can you confirm?  Related discussion around why/how app-SRE are approvers for openshift/cincinnati over [here][1].

[1]: https://github.com/openshift/cincinnati/pull/193#discussion_r365363305

Comment (vrutkovs) [reply projects/cincinnati/OWNERS:2]: Having access to this namespace might help with resolving issues on stage. I copied this from cincinnati PR owners, but its not necessary to add app-sre for this to work

**Comment by wking** [comment]
> Given this outlook I would prefer to skip the root caching for now. Instead, let's direct out attention towards unifying the build environment around the RHEL-based production builds.

This is where I'm at.  I am personally not all that familiar with Rust, and not super excited about maintaining a Rust build root for the CI cluster.  If other Cincinnati approvers want to manage this, then that's fine with me.  If ART will manage it, that's even better.  But since I'm unlikely to be a productive contributor in this space, can I be dropped from the approver set?

**Comment by LalatenduMohanty** [comment]
I am not supportive of the idea of introducing a manual step  to trigger  for updating the build root i.e. `oc command or web console triggering`. However if other Cincinnati approvers think it is fine I can live with it.

**Comment by steveeJ** [comment]
> I am not supportive of the idea of introducing a manual step to trigger for updating the build root

I'm in the same boat. I think we should propose a change to the test-platform-maintainers instead of working around it. That way others can benefit as well, and we don't have the maintenance burden of the workaround.

**Comment by vrutkovs** [comment]
Superseded by #9398

**Pull request closed without merging**